report-generator/src/main/kotlin/com/teamscale/report/util/AntPatternUtils.kt

```diff
@@ -1,16 +1,170 @@
+/*-------------------------------------------------------------------------+
+|                                                                          |
+| Copyright 2005-2011 The ConQAT Project                                   |
+|                                                                          |
+| Licensed under the Apache License, Version 2.0 (the "License");          |
+| you may not use this file except in compliance with the License.         |
+| You may obtain a copy of the License at                                  |
+|                                                                          |
+|    http://www.apache.org/licenses/LICENSE-2.0                            |
+|                                                                          |
+| Unless required by applicable law or agreed to in writing, software      |
+| distributed under the License is distributed on an "AS IS" BASIS,        |
+| WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. |
+| See the License for the specific language governing permissions and      |
+| limitations under the License.                                           |
++-------------------------------------------------------------------------*/
 package com.teamscale.report.util
 
 import java.util.regex.Pattern
 import java.util.regex.PatternSyntaxException
 
 /**
- * Wrapper around ConQAT ANT pattern utils to make it accessible from the other modules.
+ * Utility methods for dealing with Ant pattern as defined at
+ * http://ant.apache.org/manual/dirtasks.html#patterns
+ *
+ * We implement a special version where a trailing '.' can be used to only match
+ * files without file extension (i.e. file names without dot).
  */
 object AntPatternUtils {
 
     /** Converts an ANT pattern to a regex pattern.  */
     @Throws(PatternSyntaxException::class)
     fun convertPattern(antPattern: String, caseSensitive: Boolean): Pattern {
-        return org.conqat.lib.commons.filesystem.AntPatternUtils.convertPattern(antPattern, caseSensitive)
+        var normalizedAntPattern = normalizePattern(antPattern)
```

Comment:
Why can't we reuse the one from the commons anymore? is that jar not java 7 compatible?

Reply:
if we indeed can't reuse, please also copy the tests here

Reply:
That's exactly the case. I needed to replace our lib commons as it is compiled against Java 8.